src/www/myInfo.js

```diff
@@ -0,0 +1,273 @@
+        showSubject(meRes);
+        showLearningstyles(meRes);
+        showGroup(mygroupRes, meRes.name);
+    }
+
+    /**
+     * @summary Sets the title to the name of the logged in user
+     * @param {object} content the object with the logged in users informations
+     */
+    function showNameOnSite(content) {
+        const username = document.getElementById("username");
+        username.innerText = "Welcome, " + content.name;
+    }
+
+    /**
+     * @summary Prints the information about work enviroment from the user
+     * @param {object} content the object with the logged in users informations
+     */
+    function showWorkEnvironment(content) {
+        const workingEnvironmentDiv = document.getElementById("workFromHome");
+        const work = content.criteria.workingAtHome;
+        const workingEnvironment = document.createElement("p");
+        const paragraph = document.createElement("h3");
+        paragraph.innerText = "Your prefered place to work:";
+
+        clearChildren(workingEnvironmentDiv);
+
+        workingEnvironmentDiv.appendChild(paragraph);
+        workingEnvironmentDiv.appendChild(workingEnvironment);
+        workingEnvironment.innerText = workEnvironmentStringMaker(work);
+    }
+
+    /**
+     * @summary Converts a number to a string
+     * @param {number} work the number the user has sat about working environment
+     * @returns {string} Returns a string
+     */
+    function workEnvironmentStringMaker(work) {
+        let place;
+        switch (work) {
+        case 0:
+            place = "Work from home";
+            break;
+
+        case 1:
+            place = "Don't care";
+            break;
+
+        case 2:
+            place = "Work in office";
+            break;
+        }
+        return place;
+    }
+
+    /**
+     * @summary Creates the elements shown on the site about subjects the user have chossen
+     * @param {object} content the object with the logged in users informations
+     */
+    function showSubject(content) {
+        let subjectArray = content.criteria.subjectPreference.subjects.slice();
+        const subjectList = document.createElement("ul");
+        const subjectDiv = document.getElementById("subjectPreference");
+        const paragraph = document.createElement("h3");
+        paragraph.innerText = "Your subject preferences:";
+
+        subjectArray.sort((a, b) => b.score - a.score);
+        clearChildren(subjectDiv);
+        subjectDiv.appendChild(paragraph);
+
+        for (const subject of subjectArray) {
+            const subjectPart = createListItem(subject.name + ": " + (subject.score * 10).toFixed(2));
+            subjectList.appendChild(subjectPart);
+            subjectDiv.appendChild(subjectList);
+        }
+    }
+
+    /**
+     * @summary Creates a button for each file in the data folder
+     * @param {string} innerText A string
+     * @returns {HTMLElement} Returns a html li element with the string sat as innerText
```

Comment:
Der er en tilsvarende funktion i candidatelist.js please fyr begge disse to ind i en ny fil og brug den

Reply:
fixed